magenta-lib/src/main/scala/magenta/switches.scala

```diff
@@ -0,0 +1,65 @@
+// Modified version of https://github.com/guardian/guardian-management/blob/c7d99120927958757b35bea68931c4bd8690d64a/management/src/main/scala/com/gu/management/switchables.scala
```

Comment:
Copied from Guardian Management as rewriting switch logic would have involved wider changes. Not all of the switch logic from Guardian Management has been copied though, only the basic switch models.